lib/Fhp/Segment/HKTAN.php

```diff
@@ -14,64 +14,44 @@
  */
 class HKTAN extends AbstractSegment
 {
-    const NAME = 'HKTAN';
-    const VERSION = 6;
+	const NAME = 'HKTAN';
+	const VERSION = 6;
 
-    /**
-     * HKCDL constructor.
-     * @param int $version
-     * @param int $segmentNumber
-     */
-    public function __construct($version, $segmentNumber, $processID = null)
-    {
-		$data = array();
-		$data[] = 4;
-		if($processID){
-			$data = array();
-			if ($version == 6) {
-				$data[] = 2; 
-				$data[] = ""; 
-				$data[] = ""; 
-				$data[] = "";
-				$data[] = $processID;
-				$data[] = "N";
-			} else { 
-				$data[] = 2; 
-				$data[] = ""; 
-				$data[] = ""; 
-				$data[] = "";
-				$data[] = $processID;
-				$data[] = "";
-				$data[] = "N";
-			}
-		
-			#$data[] = 2;
-			#$data[] = "";
-			#$data[] = "";
-			#$data[] = "";
-			#$data[] = $processID;
-		} else {
-			$data[] = "HKIDN";
-			$data[] = "";
-			$data[] = "";
-			$data[] = "";
-			$data[] = "N";
-		}
-		
-		
-        parent::__construct(
-            static::NAME,
-            $segmentNumber,
-            $version,
-            $data
-        );
-    }
+	/**
+	 * HKCDL constructor.
+	 * @param int $version
+	 * @param int $segmentNumber
+	 */
+	public function __construct($version, $segmentNumber, $processID = null, $tanMediaName = '')
```

Comment:
Mainly refactoring large if/if/else blocks but also increasing the dataset to 12 elements to include the tan media name.

Reply:
:heavy_check_mark: